samza-core/src/test/java/org/apache/samza/processor/TestStreamProcessor.java

```diff
@@ -147,28 +181,119 @@ public void onFailure(Throwable t) {
         return null;
       }).when(mockJobCoordinator).start();
 
-    try {
-      processor.start();
-      processorListenerStart.await();
+    processor.start();
+    processorListenerStart.await();
 
-      Assert.assertEquals(SamzaContainerStatus.STARTED, processor.containerReference.getStatus());
+    Assert.assertEquals(SamzaContainerStatus.STARTED, processor.container.getStatus());
 
-      // This block is required for the mockRunloop is actually start.
-      // Otherwise, processor.stop gets triggered before mockRunloop begins to block
-      processor.runLoopStartForMain.await();
+    // This block is required for the mockRunloop is actually start.
+    // Otherwise, processor.stop gets triggered before mockRunloop begins to block
+    processor.runLoopStartForMain.await();
 
-      processor.stop();
+    processor.stop();
 
-      processorListenerStop.await();
-    } catch (InterruptedException e) {
-      e.printStackTrace();
-    }
+    processorListenerStop.await();
+
+    // Assertions on which callbacks are expected to be invoked
+    Assert.assertTrue(processorListenerState.get(ON_START));
+    Assert.assertTrue(processorListenerState.get(ON_SHUTDOWN));
+    Assert.assertFalse(processorListenerState.get(ON_FAILURE));
+  }
```

Comment:
Optional: you could respecify this as Set (or Map) equality on the expected states. It would give you slightly more contextual information in the even of a test failure (e.g. possibly detecting a second hidden failure).